### DISCUSSION TITLE
How to Create a chatflow based on Knowledgebase or Own Document Data?

### DISCUSSION THREAD

**Discussion by NathanParas** [FlowiseAI/Flowise#1708] [Q&A]
Hi Anyone here knows how to create a chatflow based on a knowledgebase like Docs, or CSV or own data then. I will create an FAQ Chatbot, if the question is not in the KB then it will response based from its own knowledge. Thanks

**Comment by Jaredude** [comment]
The marketplace is a good place to get started. Creating what you're looking to do is fairly straightforward. You'd create two separate flows:

1. Vector Upsert flow with your document loaders (text file, css, pdf, etc) attaching an appropriate text splitter. And link them to your vector node. Then load the data into your vector db. Something along the following
![Screenshot 2024-02-11 at 3 51 37 PM](https://github.com/FlowiseAI/Flowise/assets/10228950/5adb9e10-49f3-48cf-adc5-2f7b2097c3e8)

2. A chat flow with an embeddings node, vector db as retriever, chat model, and a Conversation Retrieval QA chain with a good prompt to handle what you're talking about with missing KB knowledge be handled by general model knowledge. Something along the following:
![Screenshot 2024-02-11 at 3 54 31 PM](https://github.com/FlowiseAI/Flowise/assets/10228950/7d501cd8-1419-4584-bead-d453ab2cfb45)

Hope that helps
